code/extensions/che-on-start/src/extension.ts

```diff
@@ -13,7 +13,24 @@
 import * as vscode from 'vscode';
 
 export async function activate(_context: vscode.ExtensionContext): Promise<void> {
-  const output = vscode.window.createOutputChannel('Che Project');
+
+  const output = vscode.window.createOutputChannel('Che OnStart');
+  openTerminalIfNone(output);
+  openProjectIfNone(output);
+
+}
+
+async function openTerminalIfNone(output: vscode.OutputChannel): Promise<void> {
+  // open a new terminal if there is none to always see a terminal
+  if (vscode.window.terminals.length === 0) {
+    output.appendLine('Opening a new terminal....');
```

Comment:
`output` is declared in another function - `activate`.
What `output` does it use here?

Reply:
it's using the output given in parameter initialized in activate method ?

Reply:
ahh... I overlooked it's passed through the parameter. Sorry for the noise )